src/backend/distributed/operations/repair_shards.c

```diff
@@ -321,6 +327,8 @@ citus_move_shard_placement(PG_FUNCTION_ARGS)
 	ShardInterval *shardInterval = LoadShardInterval(shardId);
 	Oid distributedTableId = shardInterval->relationId;
 
+	AcquireColocationLock(relationId, ExclusiveLock, "move placement of");
```

Comment:
We're putting up another barrier towards parallel shard moves. Kind of wondering whether we should take ShareLock here and a conflicting lock in create_distributed_table.

Reply:
I thought about this as well. But, we are planning to backport this, and using ShareLock on shard moves would change the behaviour. 

It used to throw an error when you move 2 placements of the same table, with ShareLock we'd allow.

I think to allow parallel shard moves, we'd need to change (colocationid) to something like `colocationId, shardIndex`